docs/examples/reusable-components.md

```diff
@@ -0,0 +1,262 @@
+
+```scala mdoc:js:shared
+import cats.Show
+import cats.kernel.Eq
+
+sealed trait CurrencyPair
+
+object CurrencyPair {
+  // In this example, we choose few currency pairs
+  case object EURUSD extends CurrencyPair
+  case object USDCHF extends CurrencyPair
+  case object GBPUSD extends CurrencyPair
+  case object CHFEUR extends CurrencyPair
+
+  // try to parse a string to a `CurrencyPair`
+  def fromString(s: String): Option[CurrencyPair] = s match {
+    case "EURUSD" => EURUSD.some
+    case "USDCHF" => USDCHF.some
+    case "GBPUSD" => GBPUSD.some
+    case "CHFEUR" => CHFEUR.some
+    case _        => none
+  }
+
+  // all available currency pairs
+  val all = List(EURUSD, USDCHF, GBPUSD, CHFEUR)
+
+  implicit val show: Show[CurrencyPair] = Show.show {
+    case EURUSD => "EURUSD"
+    case USDCHF => "USDCHF"
+    case GBPUSD => "GBPUSD"
+    case CHFEUR => "CHFEUR"
+  }
+
+  implicit val eq: Eq[CurrencyPair] = Eq.fromUniversalEquals
+}
+```
+
+## Actions
+
+```scala mdoc:js:shared
+sealed trait Action
+case class SetCurrencyPair(pair: CurrencyPair) extends Action
+case object MakeApiRequest extends Action
+case class SetApiResponse(response: Option[ApiResponse]) extends Action
+```
+
+## Store
+
+```scala mdoc:js:shared
+import cats.effect._
+import cats.syntax.all._
+
+object Store {
+
+  def apply[F[_]: Async]: Resource[F, ff4s.Store[F, State, Action]] =
+    ff4s.Store[F, State, Action](State()) { store =>
+      _ match {
+        case SetCurrencyPair(pair) => _.copy(currencyPair = pair) -> none
+        case SetApiResponse(response) =>
+          state =>
+            state.copy(
+              apiResponse = response,
+              numOfRequests = state.numOfRequests + 1
+            ) -> none
+        case MakeApiRequest =>
+          state =>
+            (
+              state,
+              state.ccyPairOption.some.map { case (ccy1, ccy2) =>
+                ff4s
+                  .HttpClient[F]
+                  .get[ApiResponse](
+                    s"https://api.frankfurter.app/latest?from=$ccy1&to=$ccy2"
+                  )
+                  .flatMap(response =>
+                    store.dispatch(SetApiResponse(response.some))
+                  )
+
+              }
+            )
+      }
+    }
+}
+```
+
+## View
+
+```scala mdoc:js:shared
+import cats.syntax.all._
+
+object View {
+
+  def apply[F[_]](implicit dsl: ff4s.Dsl[F, State, Action]) = {
+
+    import dsl._
+    import dsl.html._
+
+    // create a `Components` instance and import components
+    val components = new Components[F, State, Action]
+    import components._
+
+    useState { state =>
+      div(
+        customSelect[CurrencyPair](
+          CurrencyPair.fromString,
+          (_, pair) => SetCurrencyPair(pair).some,
+          CurrencyPair.all,
+          _.currencyPair
+        ),
+        div(s"Selected currency pair: ${state.currencyPair}"),
+        customButton(dsl)(
+          span("Get FX Rate"),
```

Comment:
If we are not using the button in its full generality, then we might as well make it accept a string for a label and get the benefit of passing `dsl` implicitly.